packages/model-serving/src/components/deploymentWizard/fields/DeploymentStrategyField.tsx

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import { Radio, Stack, StackItem } from '@patternfly/react-core';
+import { z } from 'zod';
+import { useAppContext } from '@odh-dashboard/internal/app/AppContext';
+
+// Schema
+export const deploymentStrategyFieldSchema = z.enum(['rolling', 'recreate']);
+
+export type DeploymentStrategyFieldData = z.infer<typeof deploymentStrategyFieldSchema>;
+
+export const isValidDeploymentStrategy = (value: unknown): value is DeploymentStrategyFieldData => {
+  return deploymentStrategyFieldSchema.safeParse(value).success;
+};
+
+// Hook
+export type DeploymentStrategyFieldHook = {
+  data: DeploymentStrategyFieldData;
+  setData: (data: DeploymentStrategyFieldData) => void;
+};
+
+export const useDeploymentStrategyField = (
+  existingData?: DeploymentStrategyFieldData,
+): DeploymentStrategyFieldHook => {
+  const { dashboardConfig } = useAppContext();
+  const hasInitializedRef = React.useRef(false);
+
+  const clusterDefault = React.useMemo(() => {
+    const strategy = dashboardConfig.spec.modelServing?.deploymentStrategy;
+    if (strategy === 'rolling' || strategy === 'recreate') {
+      return strategy;
+    }
+    return 'rolling';
+  }, [dashboardConfig.spec.modelServing?.deploymentStrategy]);
+
+  const [deploymentStrategy, setDeploymentStrategy] = React.useState<DeploymentStrategyFieldData>(
+    () => existingData || clusterDefault,
+  );
+
+  React.useEffect(() => {
+    if (!existingData && !hasInitializedRef.current) {
+      hasInitializedRef.current = true;
+      setDeploymentStrategy(clusterDefault);
+    }
+  }, [clusterDefault, existingData]);
```

Comment:
What case does this cover? If no existing data on initialization, set the strategy to default. But doesn't line 36 right above set the clusterDefault as the initial value if there's no existing data?